### ISSUE TITLE
Cannot use short node-id for detaching nomad volumes

### ISSUE THREAD

**Issue by hongkongkiwi** [hashicorp/nomad#8963]
### Nomad version
Nomad v0.12.4 (8efaee4ba5e9727ab323aaba2ac91c2d7b572d84)

(same error with v0.12.5)

### Operating system and Environment details
Linux builder0 4.15.0-112-generic 113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

### Issue
When using `nomad volume detach` we cannot use a shortform of the node-id.

For example:
`nomad volume detach builder0 "35750b4b"` is rejected with error
```
Error detaching volume: Unexpected response code: 500 (rpc error: rpc error: missing external node ID: node lookup failed: index error: UUID must be 36 characters)
```

But using `nomad detach builder0 "35750b4b-b831-5f36-6999-b6533994e02b"` is accepted.

I expect to be able to use the shortform, like it can be used in other places. Also, when using nomad volume status, only the shortform node-id is presented, so it's expected we can then use it to perform an action.

**Comment by tgross** [comment]
Hi @hongkongkiwi! Yup, that looks like a clearcut bug to me. https://github.com/hashicorp/nomad/blob/v0.12.5/command/volume_detach.go#L47 suggests we should be doing the prefix search, but that must be misconfigured somehow.

**Comment by tgross** [comment]
I've merged a fix in #9041 and that will ship in the upcoming 0.13 release. Thanks again @hongkongkiwi!
